packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -321,6 +362,10 @@ export class OdspDocumentService implements IDocumentService {
                 this.hostPolicy.sessionOptions?.unauthenticatedUserDisplayName,
             );
 
+        if (bypassCache) {
```

Comment:
I'm not sure using bypassCache is ideal. This partially comes from not ideal flow before your changes.
Basically before your changes we were trying to use existing joinSession, and we had no direct way of checking if it expired already, so we played some odds (if it did due to storage token expiry, then we would fail to connect to ordering service and would retry join session).

With a new flow, we know the time when previous joinSession results are no long valid, and thus cached value can't be used. So it would be great all the code to leverage that. Specificially, bypassCache = false path, as we might find (in between reconnects) that joinSession is still Ok (especially "read" -> "write" upgrades - same as today), but long term disconnects (like 20 min) - not really.

I'd suggest changing how we cache it. Basically, at the time of caching, we can calculate time when it expires, and either let caching layer deal with eviction, or let it never expire in cache, but make reader (this code) get the value and check if it still valid or not.

Reply:
Removed this argument.